bpo-31904: Fix test_netrc for VxWorks RTOS (#21675)
python/cpython: Doc/library/netrc.rst

--------------

This module implements an object-oriented interface to :file:`.netrc` files.

> .. availability:: Unix; not supported on VxWorks, that doesn't have home directories

Reviewer: I don't think that it's worth it to mention home directories. What do you think of this suggestion?

```suggestion
.. availability:: Unix. Not available on VxWorks.
```

Reviewer: I understand that the only issue on VxWorks is that there is no concept of user directory. But how does it prevent to use the netrc module if you specify a filename?

Maybe only skip the test but revert netrc.rst changes, to leave the documentation as it is?

Author: Yes. VxWorks users can use netrc module by specifying a filename. Reverted this.

--------------

This module implements an object-oriented interface to :file:`.netrc` files.

VxWorks has no user home directory so it doesn't support ftp auto-login. This module

> is not supported on VxWorks.

Reviewer 2: Minor: it seems wrong to make a note about a niche OS so high in the doc here.
Suggest pushing it a little below (just before the class documentation), maybe using an `availability` directive (example https://github.com/python/cpython/blob/6000087fe979705dc588a46a35da884cc0198c67/Doc/library/time.rst#functions ref https://github.com/python/cpython/blob/2d6097d027e0dd3debbabc702aa9c98d94ba32a3/Doc/library/intro.rst#notes-on-availability)

Author: @merwok Thanks for your review. Seems putting the note between the sentence a "The netrc class parses and encapsulates the netrc file format used by the Unix ftp program and other FTP clients" and sentence b "class netrc.netrc([file])" is not that correct because the sentence a is explaining the class netrc.netrc() other than the whole module. That is, this module is lack of module summary. So I add the availability as you proposed.

> .. availability:: Unix. VxWorks has no user home directory so it doesn't support ftp
>    auto-login. This module is not supported on VxWorks.

Reviewer 2: ```suggestion
.. availability:: Unix; not supported on VxWorks, that doesn't have home directories
```
